### PR TITLE
Tow 44 Created GET endpoint to get ALL opportunities

### PR DESCRIPTION
**Pull request by RYeeAnn**
Similar to getting an opportunity by a single ID, this will fetch ALL opportunities. I added additional opportunities in Django Admin so we can fetch more than one opportunity for testing purposes.

- Run python manage.py runserver and in Postman, GET:  http://127.0.0.1:8000/opportunities/

### CODE REVIEW

**Review by GagenvirGill — APPROVED** [review]

File: townhall/myapi/services.py [hunk 1]
```diff
@@ -21,9 +22,17 @@ def delete_volunteer(id: int) -> None:
...
     def get_opportunity(id: int) -> Opportunity:
-        return opportunity_dao.get_opportunity(id=id)
+        print(f"Fetching opportunity with ID: {id}")
+        opportunity = opportunity_dao.get_opportunity(id=id)
+        print(f"Fetched opportunity: {opportunity}")
+        return opportunity
+    
+    def get_opportunity_all() -> List[Opportunity]:
+        print("Fetching opportunities")
+        opportunities = Opportunity.objects.all()
```
Comment (ansel-rangers11) [inline townhall/myapi/services.py:33]: I think this should call the DAO method, service methods should never invoke DB operations like all() or get() or filter() etc

File: townhall/myapi/services.py [hunk 2]
```diff
@@ -21,9 +22,17 @@ def delete_volunteer(id: int) -> None:
         volunteer_dao.delete_volunteer(volunteer_id=id)
 
 class OpportunityServices:
-
     def get_opportunity(id: int) -> Opportunity:
-        return opportunity_dao.get_opportunity(id=id)
+        print(f"Fetching opportunity with ID: {id}")
+        opportunity = opportunity_dao.get_opportunity(id=id)
+        print(f"Fetched opportunity: {opportunity}")
```
Comment (ansel-rangers11) [inline townhall/myapi/services.py:28]: Good logging

File: townhall/myapi/views.py [hunk 3]
```diff
@@ -7,14 +7,41 @@
 from rest_framework import status
 from rest_framework.response import Response
 from .services import VolunteerServices as volunteer_services
+from .services import OpportunityServices as opportunity_services
```
Comment (ansel-rangers11) [inline townhall/myapi/views.py:10]: good whole file import, we prefer to do this than importing specific functions

**Review by ansel-rangers11 — CHANGES_REQUESTED** [review]
almost there, got a change to look at

**Comment by ansel-rangers11** [comment]
Try to merge this PR to` tow-43-get-opportunity-id` instead of `main` to result in cleaner commit history. See comment here https://github.com/AtriaCoop/townhallbackend/pull/32#discussion_r1702917482

**Pull request merged**